SingularityUI/app/views/request.coffee

```diff
@@ -78,38 +77,22 @@ class RequestView extends View
         @model.promptRemove =>
             app.router.navigate 'requests', trigger: true
 
-    runRequest: (e) =>
-        @model.promptRun (data) =>
-            # If user wants to redirect to a file after the task starts
-            if data.autoTail is 'on'
-                autoTailer = new AutoTailer({
-                    requestId: @requestId
-                    autoTailFilename: data.filename
-                    autoTailTimestamp: +new Date()
-                })
-
-                autoTailer.startAutoTailPolling()
-
-            else
+    runRequest: (e, taskId) => # If taskId is provided, rerun the task. Else run the task.
+        callback = (data) =>
+            unless data.afterStart in ['browse-to-sandbox', 'autoTail']
                 @trigger 'refreshrequest'
                 setTimeout ( => @trigger 'refreshrequest'), 2500
+        if taskId
+            task = new TaskHistory {taskId}
+            task.fetch()
```

Comment:
Unfortunately, that's not inherently true. While it is true if you're running from the request's page, you may also be running the task from the tasks page or the requests page.